docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: page
+title: Contributing a new ksqlDB client
+tagline: Contributing a new ksqlDB client
+description: Contribute a new ksqlDB client for your favorite language.
+keywords: ksqlDB, client
+---
+
+This page contains information and recommendations for how to contribute a new [ksqlDB client](index.md).
+
+Functionality
+-------------
+
+Clients should support the following functionality:
+- [Push and pull queries](#push-and-pull-queries)
+- [Terminating push queries](#terminating-push-queries)
+- [Inserting new rows of data into existing ksqlDB streams](#inserting-new-rows-of-data-into-existing-ksqldb-streams)
+- [Creation of new streams, tables, and persistent queries](#creation-of-new-streams-tables-and-persistent-queries)
+- [Terminating persistent queries](#terminating-persistent-queries)
+- [Listing existing streams, tables, topics, and queries](#listing-existing-streams-tables-topics-and-queries)
+
+### Push and pull queries ###
+
+Push and pull queries may be issued by sending requests to the [`/query-stream` server endpoint](TODO).
```

Comment:
Waiting on docs for the new server endpoints for these links.

Reply:
Are these HTTP requests? Im confused about how to interact with the server right off the bat. We probably need an opening section about how to generally communicate between the client+server.